### PR TITLE
Upgrade build to Sbt 1.4.0

### PR DESCRIPTION
**Pull request by LeeTibbert**
* This PR upgrades the Scala Native build to use Sbt 1.4.0.

  * Sbt 1.4.0 introduces mandatory checking for apparently unused
    keys.  It uses an algorithm which may or may not be accurate.
    Sbt 1.4.0 complains about the key 'crossSbtVersions' in several
    sub-projects.  Sbt says that it is unused; lore suggests that it
    is used in a subtle but critical way.  I did not have time to
    verify the lore, so I turned linting of for that key.  The situation
    is no worse than before. Yes, perhaps I missed an opportunity to
    get rid of a truly unseen key, but I did not want to inflict
    breakage on others. Not a way to win friends.

Documentation:

* No changelog entry necessary; internal build details.

* The build.properties version in docs/user/sbt.rst was updated.
    Also, a pre-existing warning was eliminated.

Testing:

  Safety -

  + Built and tested ("test-all") in debug mode using sbt 1.3.13 on
      X86_64 only . All tests pass.

  Efficacy -

  +  Interactive "sbt sbtVersion" in the project root shows 1.4.0.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -32,6 +32,14 @@ lazy val mimaSettings: Seq[Setting[_]] = Seq(
   }
 )
 
+// Sbt 1.4.0 introduces mandatory key lining.
```
Comment (ekrich) [inline build.sbt:35]: Spelling `linting`.

Comment (LeeTibbert) [reply build.sbt:35]: Fixed now, thank you. Sharp eyes.

File: build.sbt [hunk 2]
```diff
@@ -32,6 +32,14 @@ lazy val mimaSettings: Seq[Setting[_]] = Seq(
   }
 )
 
+// Sbt 1.4.0 introduces mandatory key linting.
+// The val crossSbtVersions is used in toolsSettings below.
+// toolsSettings are used by nir, test-runner, util, etc.
+// but Sbt 1.4.0 still complains about crossSbtVersions.
+// Disable the linting for it, rather than upsetting the apple cart by
+// deleting the probably essential crossSbeVersions. Minimal change.
```
Comment (lolgab) [inline build.sbt:40]: There is a typo:
```suggestion
// deleting the probably essential crossSbtVersions. Minimal change.
```

File: docs/user/sbt.rst [hunk 3]
```diff
@@ -234,9 +234,9 @@ support Scala/JVM or Scala.js if the Native portions have replacement
 code on the respective platforms.
 
 The primary purpose of this feature is to allow libraries to support
-Scala Native that need native `glue` code to operate. The current
+Scala Native that need native "glue" code to operate. The current
```
Comment (lolgab) [inline docs/user/sbt.rst:237]: Is this related?

File: docs/user/sbt.rst [hunk 4]
```diff
@@ -234,9 +234,9 @@ support Scala/JVM or Scala.js if the Native portions have replacement
...
 C interopt does not allow direct access to macro defined constants and
-functions or allow passing `struct`s from the stack to C functions.
+functions or allow passing "struct"s from the stack to C functions.
```
Comment (lolgab) [inline docs/user/sbt.rst:239]: Same here

**Review by lolgab — CHANGES_REQUESTED** [review]

File: docs/user/sbt.rst (hunk 3 above)
Comment (LeeTibbert) [reply docs/user/sbt.rst:237]: Lolgab  thank you for the careful review & the thought it took.

In the spirit of collegial discussion...

Yes, this change and the one immediatly below, takes some explaining.  They are not, IMHO, not extraneous
 "Boy Scout" or, inclusively, "Scout" Rule noise and distraction.

The proposed doc change are related to the base change in the sense that I can not get a clean trial build
of the file I touched.  That is a personal, and probably project requirement for anything I submit.

The doc build  from the current snapshot contains a number (10, 15) Warnings.  At one point, I had submitted
PRs to reduce that to zero.  I believe, for understandable reasons, those PRs are still in the queue.    

This change means that the file I did need to touch does not contribute to the general noise of the doc build.

Worthy enough for inclusion?  Please advise.

File: docs/user/sbt.rst (hunk 4 above)
Comment (LeeTibbert) [reply docs/user/sbt.rst:239]: Same as above, fixes Warning in doc build in a file I touched.

**Review by lolgab — APPROVED** [review]
LGTM 👍

**Comment by LeeTibbert** [comment]
Once again, Travis thinks I have not signed the CLA.   After all the other tests finish, I will do a "git commit --amend" to 
start another run.  It is almost certainly either a bad network link or overworked CPUs at one end or the other.

**Comment by LeeTibbert** [comment]
Superseded by PR #1956, which moves Sbt to version 1.4.1.
